wazo_confd/plugins/user/resource.py

```diff
@@ -1,34 +1,161 @@
+from wazo_confd.plugins.user_fallback.resource import UserFallbackList
 
 from .schema import (
     UserDirectorySchema,
-    UserSchema,
+    UserListSchema,
     UserSchemaNullable,
     UserSummarySchema,
+    UserPutSchema,
 )
+from .resource_sub import UserForwardList
+
+logger = logging.getLogger(__name__)
 
 
 class UserList(ListResource):
 
     model = User
     schema = UserSchemaNullable
+    _list_schema = UserListSchema
     view_schemas = {'directory': UserDirectorySchema, 'summary': UserSummarySchema}
 
+    def __init__(
+        self,
+        user_service,
+        line_service,
+        user_line_service,
+        endpoint_custom_service,
+        endpoint_sip_service,
+        extension_line_service,
+        extension_service,
+        line_endpoint_custom_association_service,
+        line_endpoint_sip_association_service,
+        line_endpoint_sccp_association_service,
+        endpoint_sccp_service,
+        voicemail_service,
+        user_forward_service,
+        user_fallback_service,
+        user_voicemail_service,
+        wazo_user_service,
+        endpoint_custom_dao,
+        endpoint_sccp_dao,
+        line_dao,
+        user_dao,
+        sip_dao,
+        transport_dao,
+        voicemail_dao,
+    ):
+        super().__init__(user_service)
+        self._line_list_resource = LineList(
+            line_service,
+            endpoint_custom_service,
+            endpoint_sip_service,
+            extension_line_service,
+            extension_service,
+            line_endpoint_custom_association_service,
+            line_endpoint_sip_association_service,
+            line_endpoint_sccp_association_service,
+            endpoint_sccp_service,
+            endpoint_custom_dao,
+            endpoint_sccp_dao,
+            line_dao,
+            sip_dao,
+            transport_dao,
+        )
+        self._user_line_item_resource = UserLineItem(
+            user_line_service, user_dao, line_dao
+        )
+        self._user_voicemail_item_resource = UserVoicemailItem(
+            user_voicemail_service,
+            user_dao,
+            voicemail_dao,
+        )
+        self._user_voicemail_list_resource = UserVoicemailList(
+            user_voicemail_service,
+            voicemail_service,
+            user_dao,
+            voicemail_dao,
+        )
+        self._voicemail_item_resource = VoicemailItem(voicemail_service)
+        self._wazo_user_service = wazo_user_service
+        self._user_forward_list_resource = UserForwardList(user_forward_service)
+        self._user_fallback_list_resource = UserFallbackList(
+            user_fallback_service, user_dao
+        )
+
     def build_headers(self, user):
         return {'Location': url_for('users', id=user.id, _external=True)}
 
     @required_acl('confd.users.create')
     def post(self):
         return super().post()
 
+    def _post(self, body):
+        # forwards payload will be modified by the PUT body
```

Comment:
why is it before `body = self.schema().load(body)` ?

Reply:
The schema forwards schema modifies the data. If I get the forwards after the `load` the forwards resource will try to modify the schema again. This returns a `400`